Create wr-pci-leds example

huitmj opened the pull request:
add WR PCI Leds device to system build steps
add barebone example PCI LEDs device
added documentation for wr-pci-leds specification

huitmj committed:
Create wr-pci-leds example

add WR PCI Leds device to system build steps
add barebone example PCI LEDs device
added documentation for wr-pci-leds specification

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+#include "hw/pci/pci_device.h"
+
+#define TYPE_MY_DEVICE "wr-pci-leds"
+OBJECT_DECLARE_SIMPLE_TYPE(MyDeviceState, MY_DEVICE)
```
probably ok since this is just an example implementation, but since the file name and TYPE_##MODULE_OBJ_NAME are both of wr-pci-leds nomenclature, i think the device state structure should also be named accordingly.

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+#define TYPE_MY_DEVICE "wr-pci-leds"
+OBJECT_DECLARE_SIMPLE_TYPE(MyDeviceState, MY_DEVICE)
+
+typedef struct MyDeviceState
```
same comment as above, and throughout the file, the device state structure should probably be named consistently with the device type

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+}
+type_init(my_device_register_types)
+
+/* Notes:
```
No longer needed notes?

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+};
+static void pci_my_device_realize(PCIDevice *dev, Error **errp)
+{
+    printf(">>\t<QEMU> Hello this is my-device PCI device\n");
```
to prevent these from being interleaved with the driver test output, perhaps these can be qemu log messages that we can redirect to file using "-D <filename>" option on the qemu command line, rather than requiring the tab indent

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+{
+    printf(">>\t<QEMU> Hello this is my-device PCI device\n");
+    MyDeviceState *s = MY_DEVICE(dev);
+    memory_region_init_io(&s->bar, OBJECT(s), &my_device_ops, s, "my-device", 32);
```
s/my-device/wr-pci-leds

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+    printf(">>\t<QEMU> Write to %lu with %lu %u\n",addr, data, size);
+    MyDeviceState* s = opaque;
+    switch (addr) {
+        case 0: s->led0=data; break;
```
style nit - case body should have new line per statement and indented

ho28 commented on hw/misc/wr-pci-leds.c:
```diff
@@ -0,0 +1,80 @@
+    PCIDevice parent_obj;
+    MemoryRegion bar;
+
+    int32_t led0,led1;
```
should these members be uint64_t to match the assignments in my_device_write()?

ho28 reviewed (commented):
Thank you for the contribution James! I left several comments, but if you prefer, we can merge this change as-is, and I can address my comments in a follow-up change.

huitmj committed:
Update wr-pci-leds.c

align led private data type with write data type and remove notes that no longer needed
